src/Controller/Component/AuthComponent.php

```diff
@@ -403,10 +406,10 @@ protected function _isLoginAction(Controller $controller)
      * Handle unauthorized access attempt
      *
      * @param \Cake\Controller\Controller $controller A reference to the controller object
-     * @return \Cake\Http\Response
+     * @return \Cake\Http\Response|null
      * @throws \Cake\Http\Exception\ForbiddenException
      */
-    protected function _unauthorized(Controller $controller)
+    protected function _unauthorized(Controller $controller): ?Response
```

Comment:
Does this need to nullable if the method sometimes throws an exception?

Reply:
No, it should always be either Response, or the exception to bail early.